### PR TITLE
HHH-11173 - Fix text serialization of uninitialized lazy attributes

### PR DESCRIPTION
**Pull request by barreiro**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/type/CollectionType.java [hunk 1]
```diff
@@ -194,21 +195,21 @@ public String toLoggableString(Object value, SessionFactoryImplementor factory)
 				}
 			}
 		}
```
Comment (gbadner) [inline hibernate-core/src/main/java/org/hibernate/type/CollectionType.java]: IIUC, by removing:

`if ( !Hibernate.isInitialized( value ) ) {
     return "<uninitialized>";
}`

I believe that `renderLoggableString` will initialize the collection if was uninitialized.

File: hibernate-core/src/main/java/org/hibernate/type/ComponentType.java [hunk 2]
```diff
@@ -472,7 +473,12 @@ public String toLoggableString(Object value, SessionFactoryImplementor factory)
 		Map<String, String> result = new HashMap<>();
 		Object[] values = getPropertyValues( value, entityMode );
```
Comment (gbadner) [inline hibernate-core/src/main/java/org/hibernate/type/ComponentType.java]: I'm not sure how lazy embedded values work. Can the embedded value as a whole be lazy? If so, would calling `getPropertyValues` initialize the lazy value? If this is not an issue, the following looks OK to me.

**Comment by gbadner** [comment]
A general comment, `LazyPropertyInitializer.UNFETCHED_PROPERTY.toString()` returns "<lazy>". I see that "<uninitialized>" is used instead. If this is only used for logging, then it probably doesn't matter.

**Comment by barreiro** [comment]
The main change is on `TypeHelper#toLoggableString`. The ones on specific types are just safeguards. 

In light of that, and to answer your comments directly, if a collection is not initialized then `CollectionType#getLoggableString` will not be called in the first place. 

For embedded types is a bit different, but your assumption is right: `getPropertyValues` does not trigger the load of uninitialized attributes.

Finally, can either have `<lazy>` or `<uninitialized>`. As you pointed out, it's only for logging. I kept it as is.

**Comment by gbadner** [comment]
There are other places where `Type#toLoggableString` is called. Most involve IDs and versions, which (obviously) wouldn't involve an uninitialized collection.

`EntityPrinter#toString(String, Object)` can be called when logging entities in the `PersistenceContext` after flushing. We want to be sure that no collections end up getting initialized in the process of being logged.

**Comment by gbadner** [comment]
I ended up adding a change `CollectionType#renderLoggableString` to return '<uninitialized>' when collection is uninitialized.

I did not change `EntityPrinter#toString` to use `TypeHelper#toLoggableString`, since `EntityPrinter#toString` excludes properties with a name that start with '_'.  I *think* such properties are "virtual" properties, but I'm not sure.

**Pull request closed without merging**